src/lib/core/overlay/overlay-directives.ts

```diff
@@ -50,15 +52,51 @@ export class OverlayOrigin {
  * Directive to facilitate declarative creation of an Overlay using a ConnectedPositionStrategy.
  */
 @Directive({
-  selector: '[connected-overlay]'
+  selector: '[connected-overlay]',
+  exportAs: 'connectedOverlay'
 })
-export class ConnectedOverlayDirective implements OnInit, OnDestroy {
+export class ConnectedOverlayDirective implements OnDestroy {
   private _overlayRef: OverlayRef;
   private _templatePortal: TemplatePortal;
+  private _open = false;
+  private _hasBackdrop = false;
+  private _backdropSubscription: Subscription;
 
   @Input() origin: OverlayOrigin;
   @Input() positions: ConnectionPositionPair[];
 
+  /** The width of the overlay panel. */
+  @Input() width: number | string;
+
+  /** The height of the overlay panel. */
+  @Input() height: number | string;
+
+  /** The custom class to be set on the backdrop element. */
+  @Input() backdropClass: string;
+
+  /** Whether or not the overlay should attach a backdrop. */
+  @Input()
+  get hasBackdrop() {
+    return this._hasBackdrop;
+  }
+
+  set hasBackdrop(value: any) {
+    this._hasBackdrop = value === '' || (!!value && value !== 'false');
+  }
+
+  @Input()
+  get open() {
+    return this._open;
+  }
+
+  set open(value: boolean) {
+    value ? this._attachOverlay() : this._detachOverlay();
+    this._open = value;
+  }
+
+  /** Event emitted when the backdrop is clicked. */
+  @Output() backdropClick = new EventEmitter();
```

Comment:
Yeah, I think it does. It's not throwing, so maybe it's implicitly typed to `EventEmitter<void>` when set this way?